scripts/figen_class.py

```diff
@@ -39,14 +65,10 @@ def extract_middle_percent(self, data: pd.DataFrame, start: float, last: float):
         """
         scaler = StandardScaler()
         data_scaled = scaler.fit_transform(data.values)
-        kde = KernelDensity(kernel="gaussian", bandwidth=0.5).fit(
-            data_scaled
-        )  ##TODO: 계산이 안터지도록 하기, gmm으로 변경
-        log_prob = kde.score_samples(data_scaled)
-        prob = np.exp(log_prob)
-        threshold_low, threshold_high = np.percentile(prob, [start, last])
-        mask = np.logical_and(prob >= threshold_low, prob <= threshold_high)
-        data_middle = data[mask]
+
+        # 분위수 기반으로 데이터 추출
+        self_kde = self.extract_kernel(data_scaled, start, last)
```

Comment:
```suggestion
        kde_condition = self.extract_kernel(data_scaled, start, last)
```
등으로 변수명을 변경해주시면 좋을거 같아요. (self 는 삭제)